.travis.yml

```diff
@@ -1,15 +1,56 @@
+---
+sudo: required
+services:
+  - docker
 language: python
-
+python:
+  - "2.7"
+  - "3.4"
+env:
```

Comment:
if you use matrix environments you can avoid all of the if statements below.
```
env:
  matrix:
    OS:
      - centos
      - ubuntu
   OS_VERSION:
      - 6
      - 7
      - trusty
      - xenial
```
Then you can [exclude illogical elements of the matrix](https://docs.travis-ci.com/user/customizing-the-build/#build-matrix) : 

```
exclude:
  - OS: centos
    OS_VERSION:
    - xenial
    - trusty
  - OS: ubuntu
    OS_VERSION:
    - 6
    - 7
```

Reply:
Oh, nice!

Reply:
But as I no more have a matrix it will be for next time